src/main/java/org/breedinginsight/api/v1/controller/OntologyController.java

```diff
@@ -135,4 +143,82 @@ public HttpResponse<Response<DataResponse<Trait>>> revokeOntology(
             return HttpResponse.status(HttpStatus.NOT_FOUND, e.getMessage());
         }
     }
+
+    /**
+     * Scoped under the program that is subscribing. Accept a shared ontology.
+     *
+     * @param programId -- Program that is subscribing to an ontology
+     * @param sharingProgramId -- Program that has shared its ontology.
+     * @return
+     */
+    @Put("/programs/{programId}/ontology/subscribe/{sharingProgramId}")
+    @Produces(MediaType.APPLICATION_JSON)
+    @AddMetadata
+    @ProgramSecured(roles = {ProgramSecuredRole.BREEDER})
+    public HttpResponse<Response<SubscribedProgram>> subscribeOntology(
+            @PathVariable UUID programId, @PathVariable UUID sharingProgramId) {
+        try {
+            SubscribedProgram shareRequest = ontologyService.subscribeOntology(programId, sharingProgramId);
+            Response<SubscribedProgram> response = new Response(shareRequest);
+            return HttpResponse.ok(response);
+        } catch (UnprocessableEntityException e) {
+            return HttpResponse.status(HttpStatus.UNPROCESSABLE_ENTITY, e.getMessage());
+        } catch (DoesNotExistException e) {
+            return HttpResponse.status(HttpStatus.NOT_FOUND, e.getMessage());
+        }
+    }
+
+    /**
+     * Scoped under the program that is subscribing. Unsubscribe from a shared ontology.
+     *
+     * @param programId -- Program that is unsubscribing from an ontology
+     * @param sharingProgramId -- Program that has shared its ontology.
+     * @return
+     */
+    @Delete("/programs/{programId}/ontology/subscribe/{sharingProgramId}")
+    @Produces(MediaType.APPLICATION_JSON)
+    @ProgramSecured(roles = {ProgramSecuredRole.BREEDER})
+    public HttpResponse unsubscribeOntology(
+            @PathVariable UUID programId, @PathVariable UUID sharingProgramId) {
+        try {
+            ontologyService.unsubscribeOntology(programId, sharingProgramId);
+            return HttpResponse.ok();
+        } catch (UnprocessableEntityException e) {
+            return HttpResponse.status(HttpStatus.UNPROCESSABLE_ENTITY, e.getMessage());
+        } catch (DoesNotExistException e) {
+            return HttpResponse.status(HttpStatus.NOT_FOUND, e.getMessage());
+        }
+    }
+
+    /**
+     * Retrieves subscription options for programs that have shared their ontology with the requesting program.
+     * Will indicate whether the program has subscribed to a given ontology or not.
+     *
+     * @param programId -- Program request information
+     * @return
+     * {
+     *     programId,   -- Program that owns the ontology the request program is subscribed to.
```

Comment:
Small thing, should it be "Program that owns the ontology the request program may subscribe to." since this retrieves subscription options?